jax/_src/pjit.py

```diff
@@ -1757,6 +1757,20 @@ def _resolve_wsc_args(axis_resources, shardings):
 # period is finished. The deprecation period expires 3 months from Feb 13, 2023.
 def with_sharding_constraint(x, shardings=UNSPECIFIED,
                              axis_resources=UNSPECIFIED):
+  """Mechanism to constrain the sharding of an Array inside a jitted computation
+
+  This is a strict constraint for the GSPMD partitioner and not a hint. For examples
+  of how to use this function, see `Distributed arrays and automatic parallelization`_.
+
+  Args:
+    x: PyTree of jax.Arrays which will have their shardings constrainted
+    shardings: PyTree of jax.sharding.Sharding instances specifying the sharding
```

Comment:
This can be PartitionSpec too: You should copy paste this here: https://github.com/google/jax/blob/main/jax/_src/pjit.py#L628-L641

Reply:
This seems like a lot of information to put into a parameter description. Can we describe this somewhere else in the docs, so that these docstrings can refer to a canonical description rather than pasting the same information?

Reply:
I replaced this description with a reference to the other doc.

Reply:
The doc does not tell all the valid values. pjit and jit both have those things so it should be fine to put it here too.

Reply:
OK, I cross-referenced to the pjit docs. WDYT?